test/function/samples/emit-file/deduplicate-binary-asset/_config.js

```diff
@@ -0,0 +1,18 @@
+const source = Buffer.from('abc');
+
+module.exports = {
```

Comment:
As I see it, this test does not really have any assertions and a `function` test may not be the best fit for asset deduplication as it does not verify `_expected`. Why not just extend the existing `chunking-form` test as it was before Rollup 3: https://github.com/rollup/rollup/blob/69ff4181e701a0fe0026d0ba147f31bc86beffa8/test/chunking-form/samples/emit-file/deduplicate-assets/_config.js
We had a test for buffer deduplication, so it should just be green again

Reply:
You could also include cross Buffer/string deduplication as well in the test

Reply:
Ah, nice! I missed the `deduplicate-assets` test before. Extended it in [rollup/rollup@`9dbffe1` (#4712)](https://github.com/rollup/rollup/pull/4712/commits/9dbffe152fb0fe90ab2768e4843c076e84507d4d) and also added a different buffer source for completeness.

I also didn't know that Rollup 2 was already deduplicating binary sources. So then there is even less of a reason for the option if this was already how things worked before. Is my understanding correct that cross-buffer/string deduplication didn't happen in Rollup 2 though?

Reply:
>  Is my understanding correct that cross-buffer/string deduplication didn't happen in Rollup 2 though?

Yes, that is indeed a new feature!